git-config/src/parse.rs

```diff
@@ -0,0 +1,139 @@
+use crate::{file, spanned};
+use dangerous::{Bytes, BytesReader, Error, Expected, Input};
+
+fn config(bytes: &[u8]) -> Result<Vec<file::Token>, Expected<'_>> {
+    fn config<'i, E>(input: Bytes<'i>, r: &mut BytesReader<'i, E>) -> Result<Vec<file::Token>, E>
+    where
+        E: Error<'i>,
+    {
+        let mut tokens = Vec::new();
+        skip_whitespace_or_comment(r, ConsumeTo::NextToken).map(|section| {
+            tokens.push(spanned::Comment(
+                dangerous::input(section)
```

Comment:
In theory this conversion would have to be done for each token we parse here. Certainly this can be sugared up a bit more to feel less cluttered, but might also be a case for just using `zc`.
After all, the `File` type really is an `zc` type, owning data and keeping offsets into the data. The data will always be read into memory I think, so that would match too (as opposed to memory mapping them).